TACC/Core-Portal
# Task/fp 341  ui (no ux)

## Overview: ##

UI (no functional change, no UX wiring, despite branch name) for PR #87.

## PR Status: ##


## Related Jira tickets: ##

* __[FP-341](https://jira.tacc.utexas.edu/browse/FP-341)__
* [FP-508](https://jira.tacc.utexas.edu/browse/FP-508)
* _(disabled support for) [FP-563](https://jira.tacc.utexas.edu/browse/FP-563)_

## Summary of Changes: ##

### Relevant
- __New__: Include the `DataFilesSearchbar` component.
    - _[FP-508](https://jira.tacc.utexas.edu/browse/FP-508): Use the `DropdownSelector` component._
- __New__: Convert `DataFiles.scss` and `DataFilesSearchbar.scss` SCSS to CSS Modules.
- __New__: Add ([unused](https://jira.tacc.utexas.edu/browse/FP-563)) status message, dropdown filter placeholder, and "Back to All Files" link.
- __New__: Update `DataFilesSearchbar` tests.
    - _Replace `keyDown`/`click` events with single `submit` event._
    - _Add "has expected elements" test._
    - _Get some elements by role (less brittle than text)._
- _Minor_: Organize button styles and hierarchy of overrides.

## Testing Steps: ##
1. Load Data Files.
1. Ensure _new_ elements are present and laid out according to design (whose dimensions must be multiplied by 1.2).

## UI Photos:

Narrow
![FP-341 FP-508 UI - Narrow](https://user-images.githubusercontent.com/62723358/89205405-4d232a80-d57d-11ea-8d17-ce9ab93ac3f7.png)

Wide
![FP-341 FP-508 UI - Wide](https://user-images.githubusercontent.com/62723358/89205412-514f4800-d57d-11ea-864a-a6ca86c115dc.png)

Archive
- [FP-341 UI - Narrow](https://user-images.githubusercontent.com/62723358/89089290-3d7bca00-d362-11ea-82da-a24fd9d5e002.png)
- [FP-341 UI - Wide](https://user-images.githubusercontent.com/62723358/89088537-a3665280-d35e-11ea-9595-56af3699ca49.png)


## Notes: ##

### Known Issues

1. Search is __not__ integrated with data listing.
1. Filter is __not__ integrated with data listing.
